File: Programmers/[43238] 입국 심사.py
def solution(n, times):
    left = 1
    right = n * max(times)

    while left < right:
        mid = (left + right) // 2
        cnt = 0

        for time in times:
            cnt += mid // time

        if cnt >= n:
            right = mid
        else:
            left = mid + 1

    return left


if __name__ == "__main__":
    n, times = 6, [7, 10]
    print(solution(n, times))

File: Programmers/[43238] 입국심사.java
class Solution {
    public long solution(int n, int[] times) {

        long left = 0;
        long right = (long) n * times[times.length - 1];
        while (left + 1 < right) {
            long mid = (long) (left + right) / 2;

            long total = 0;
            for (int time : times) {
                total += (long) mid / time;
            }
            
            if (total >= n) {
                right = mid;
            } else {
                left = mid;
            }
        }
        
        return right;
    }
}